sway-lsp/src/utils/keyword_docs.rs

```diff
@@ -0,0 +1,850 @@
+            /// A value of type [`bool`] representing logical **true**.
+            ///
+            /// Logically `true` is not equal to [`false`].
+            ///
+            /// ## Control structures that check for **true**
+            ///
+            /// Several of Sway's control structures will check for a `bool` condition evaluating to **true**.
+            ///
+            ///   * The condition in an [`if`] expression must be of type `bool`.
+            ///     Whenever that condition evaluates to **true**, the `if` expression takes
+            ///     on the value of the first block. If however, the condition evaluates
+            ///     to `false`, the expression takes on value of the `else` block if there is one.
+            ///
+            ///   * [`while`] is another control flow construct expecting a `bool`-typed condition.
+            ///     As long as the condition evaluates to **true**, the `while` loop will continually
+            ///     evaluate its associated block.
+            ///
+            ///   * [`match`] arms can have guard clauses on them.
+            mod true_keyword {}
+        };
+
+        let false_keyword: ItemMod = parse_quote! {
+            /// A value of type [`bool`] representing logical **false**.
+            ///
+            /// `false` is the logical opposite of [`true`].
+            ///
+            /// See the documentation for [`true`] for more information.
+            mod false_keyword {}
+        };
+
+        let break_keyword: ItemMod = parse_quote! {
+            /// Exit early from a loop.
+            ///
+            /// When `break` is encountered, execution of the associated loop body is
+            /// immediately terminated.
+            ///
+            /// ```sway
+            /// let mut x = 0;
+            ///
+            /// for x < 100 {
+            ///     if x > 12 {
+            ///         break;
+            ///     }
+            ///     x += 1;
+            /// }
+            ///
+            /// assert(x == 12);
+            /// ```
+            mod break_keyword {}
+        };
+
+        let continue_keyword: ItemMod = parse_quote! {
+            /// Skip to the next iteration of a loop.
+            ///
+            /// When `continue` is encountered, the current iteration is terminated, returning control to the
+            /// loop head, typically continuing with the next iteration.
+            ///
+            /// ```sway
+            /// // Printing odd numbers by skipping even ones
+            /// for number in 1..=10 {
+            ///     if number % 2 == 0 {
+            ///         continue;
+            ///     }
+            ///     log(number);
+            /// }
+            /// ```
+            mod continue_keyword {}
+        };
+
+        // TODO
+        let str_keyword: ItemMod = parse_quote! {
+            mod str_keyword {}
+        };
+
+        // TODO
+        let for_keyword: ItemMod = parse_quote! {
```

Comment:
Are the rust docs for `for` different from sway?

Reply:
Yeah, we don't have support for iterators in sway yet so we don't have `for` loops. The only place `for` is used is when implementing traits for a type.